employeeApp/src/components/EmployeeForm/EmployeeForm.tsx

```diff
@@ -0,0 +1,325 @@
+import { useEffect, useState } from "react";
+import "./EmployeeForm.scss";
+import { Link } from "react-router-dom";
+import EmployeeService from "../../services/EmployeeService";
+import { useNavigate } from "react-router-dom";
+import { useParams } from "react-router";
+
+interface props {
+  employeeDetails: any;
+}
+
+const EmployeeForm = ({ employeeDetails }: props) => {
+  const [firstName, setFirstName] = useState("");
+  const [middleName, setMiddleName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [emailId, setEmailId] = useState("");
+  const [mobileNum, setMobileNum] = useState("");
+  const [address, setAddress] = useState("");
+  const [contractType, setContractType] = useState("");
+  const [startDate, setStartDate] = useState("");
+  const [finishedDate, setFinishedDate] = useState("");
+  const [workType, setWorkType] = useState("");
+  const [workHours, setWorkHours] = useState("");
+  const navigate = useNavigate();
+  const { id } = useParams();
+
+  useEffect(() => {
+    setFirstName(employeeDetails?.firstName);
+    setMiddleName(employeeDetails?.middleName);
+    setLastName(employeeDetails?.lastName);
+    setEmailId(employeeDetails?.emailId);
+    setMobileNum(employeeDetails?.mobileNum);
+    setAddress(employeeDetails?.address);
+    setContractType(employeeDetails?.contractType);
+    setStartDate(employeeDetails?.startDate);
+    setFinishedDate(employeeDetails?.finishedDate);
+    setWorkType(employeeDetails?.workType);
+    setWorkHours(employeeDetails?.workHours);
```

Comment:
loads of state, he not sure if you'll have enough time, but this is why I recommended using useForm, makes for way nicer components, lets have a chat about it

Reply:
reimplemented code to use useForm instead of useState